tests/mocha/gesture_test.js

```diff
@@ -11,77 +11,88 @@
+    var gesture = gestureWorkspace.currentGesture_;
+    chai.assert.exists(gesture, 'Gesture exists after pointerdown.');
+    var isFieldClickSpy = sinon.spy(gesture, 'isFieldClick_');
+
+    dispatchPointerEvent(eventTarget, 'pointerup');
+    dispatchPointerEvent(eventTarget, 'click');
+
+    sinon.assert.called(isFieldClickSpy);
+    chai.assert.isTrue(isFieldClickSpy.alwaysReturned(isFieldClick));
+  }
+
+  function getTopFlyoutBlock(flyout) {
+    return flyout.workspace_.topBlocks_[0];
+  }
 
   setup(function() {
     sharedTestSetup.call(this);
-    this.workspace = new Blockly.Workspace();
-    this.e = {};
+    defineBasicBlockWithField();
+    var toolbox = document.getElementById('gesture-test-toolbox');
+    this.workspace = Blockly.inject('blocklyDiv', {toolbox: toolbox});
   });
 
   teardown(function() {
     sharedTestTeardown.call(this);
+    delete Blockly.Block['test_field_block'];
   });
 
   test('Constructor', function() {
-    var gesture = new Blockly.Gesture(this.e, this.workspace);
-    chai.assert.equal(gesture.mostRecentEvent_, this.e);
+    var e = { id: 'dummy_test_event'};
+    var gesture = new Blockly.Gesture(e, this.workspace);
+    chai.assert.equal(gesture.mostRecentEvent_, e);
     chai.assert.equal(gesture.creatorWorkspace_, this.workspace);
   });
 
   test('Field click - Click in workspace', function() {
-    var block = new Blockly.Block(this.workspace);
-    var field = new Blockly.Field();
-    field.setSourceBlock(block);
-    field.showEditor_ = function() {};
-    var gesture = new Blockly.Gesture(this.e, this.workspace);
-    gesture.setStartField(field);
-
-    var isFieldClick = gesture.isFieldClick_();
-    chai.assert.isTrue(isFieldClick);
-  });
+    var block = this.workspace.newBlock('test_field_block');
+    block.initSvg();
+    block.render();
 
-  function gestureIsFieldClick_InFlyoutHelper(flyout, expectedResult){
-    // Assign workspace flyout
-    this.workspace.flyout_ = flyout;
-    // Create a Field inside of a Block
-    var block = new Blockly.Block(this.workspace);
-    var field = new Blockly.Field();
-    field.setSourceBlock(block);
-    field.showEditor_ = function() {};
-    // Create gesture from the flyout
-    var gesture = new Blockly.Gesture(this.e, this.workspace.flyout_);
-    // Populate gesture with click start information
-    gesture.setStartField(field);
-    gesture.setStartFlyout_(this.workspace.flyout_);
-
-    var isFieldClick = gesture.isFieldClick_();
-    chai.assert.equal(isFieldClick, expectedResult);
-  }
+    testGestureIsFieldClick(block, true);
+  });
 
   test('Field click - Auto close flyout', function() {
-    var flyout = new Blockly.VerticalFlyout(new Blockly.Options({}));
-    flyout.autoClose = false;
-    gestureIsFieldClick_InFlyoutHelper.call(this, flyout, true);
+    var flyout = this.workspace.flyout_;
+    chai.assert.exists(this.workspace.flyout_,
+        'Precondition: missing flyout');
+    flyout.autoClose = true;
```

Comment:
This is flyout.autoClose = false in the original. Was this meant to be changed?

Reply:
Yes, in the original the flyout autclose and always open tests were the same due to copy paste error. This fixes that.